ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,128 @@ actor_system_config:
+
+Example of the `memory_controller_config` section with a specified hard memory limit:
+
+```yaml
+memory_controller_config:
+  hard_limit_bytes: 16106127360
+```
+
+### Soft memory limit {#soft-memory-limit}
+
+The soft memory limit specifies a dangerous threshold that should not be exceeded by {{ ydb-short-name }} process under normal circumstances.
+
+If the soft limit is exceeded, {{ ydb-short-name }} gradually reduces the shared cache size to zero. Therefore, more database nodes should be added to the cluster as soon as possible, or per-component memory limits should be reduced.
+
+### Target memory utilization {#target-memory-utilization}
+
+The target memory utilization specifies a threshold for {{ ydb-short-name }} process memory usage that is considered optimal.
+
+Flexible cache sizes are calculated according to their limit thresholds to keep process consumption around this value.
+
+For example, in a database that consumes a little memory on query execution, caches consume memory around this threshold, and other memory stays free. If query execution consumes more memory, caches start to reduce their sizes to their minimum threshold.
+
+### Per-component memory limits
+
+There are two different types of components inside {{ ydb-short-name }}.
+
+The first type components, or cache components, acts like caches, for example storing the last recently used data. Each of cache components has minimum and maximum memory limit thresholds, allowing them to change their capacity dynamically based on current {{ ydb-short-name }} process consumption.
+
+The second type components, or activity components, allocate memory for specific activities, such as query execution or the [compaction](../../concepts/glossary.md#compaction) process. Each of activity components has a fixed memory limit. There is also an additional total memory limit for these activities, from which they attempt to consume the needed memory.
+
+Many other auxiliary components and processes operate alongside the {{ ydb-short-name }} process, consuming memory. Currently, these components do not have any memory limits.
+
+#### Cache components memory limits
+
+Cache components are:
+
+- Shared cache
+- MemTable
+
+Each of cache components' limits is dynamically recalculated every second so that each component consumes memory proportionally to its limit thresholds, and the total consumed memory stays around the target memory utilization.
+
+Cache components' minimum memory limit threshold isn't reserved, meaning that the memory remains available until it is actually used. However, once this memory is filled, the components typically retain the data, operating within their current memory limit. Consequently, the sum of cache components minimum memory limits is expected to be less than the target memory utilization.
+
+If needed, both the minimum and maximum thresholds should be overridden; otherwise, a missing threshold will have a default value.
```

Comment:
```suggestion
#### Cache components memory limits

The cache components include:

- Shared cache
- MemTable

Each cache component's limits are dynamically recalculated every second to ensure that each component consumes memory proportionally to its limit thresholds while the total consumed memory stays close to the target memory utilization.

The minimum memory limit threshold for cache components isn't reserved, meaning the memory remains available until it is actually used. However, once this memory is filled, the components typically retain the data, operating within their current memory limit. Consequently, the sum of the minimum memory limits for cache components is expected to be less than the target memory utilization.

If needed, both the minimum and maximum thresholds should be overridden; otherwise, any missing threshold will have a default value.
```